### PR TITLE
This month in Linebender, November 2024 edition!

### PR DESCRIPTION
**Pull request by waywardmonkeys**

### CODE REVIEW

**Comment by waywardmonkeys** [comment]
Rendered: https://github.com/linebender/linebender.github.io/blob/blog/tmil-11/content/blog/2024-12-03-tmil-11/index.md

File: content/blog/2024-12-03-tmil-11/index.md [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+This month's update contains some surprises, even to us!
+
+## Color
+
+When we published the update for [October, 2024](/blog/tmix-10/), we had already been discussing starting a new color-focused crate, but hadn't yet created a repository.
+
+We may have a dedicated blog post about the Color crate in the future and laying out some of the motivations, but we'll go over it a bit here.
+
+Linebender, among other things, believes in helping build solid foundations for people building user interfaces and working in graphics, in the Rust programming language (and on GPUs).
+Upon seeing a problem, we would rather produce a focused crate that solves that problem and that can be used by others within the Rust ecosystem.
```
Comment (tomcur) [inline content/blog/2024-12-03-tmil-11/index.md:17]: Language nitpick: the "rather" here is currently hanging. Maybe "we like to produce ...", or maybe you had different ideas about this sentence.

Comment (waywardmonkeys) [reply content/blog/2024-12-03-tmil-11/index.md]: Reworded slightly.

**Review by DJMcNab — COMMENTED** [review]
We have generally tried to scope these quite tightly to "what happened in the previous month", whereas this update seems quite focused on "we have big plans for December".

I'm inclined to be cautious about making big release announcements before we do them; I think I'm really taking umbrage with the "we will" phrasing, when in my mind there's a decent (~45%?) chance that "we will" in fact not.

File: content/blog/2024-12-03-tmil-11/index.md [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+* [kurbo#390][] Reduce number of operations in `Triangle::circumscribed_circle`.
+* [kurbo#399][] Implement `Sum` for `Vec2`.
+
+Internally, we're talking about the future direction of Kurbo and how we can improve upon it as a foundational crate for curves, shapes, and math.
+Hopefully we'll be writing about that in more detail within the next couple of months.
+
+## Peniko
+
+Peniko will have a big release within the next month, once the new release of Color is out, that will remove the old `peniko::Color` type and replace it with the new create.
+This release will be the start of a path towards wider color gamut support throughout the Linebender ecosystem.
```
Comment (xStrom) [inline content/blog/2024-12-03-tmil-11/index.md:51]: Less about the future, more about November.

```suggestion
We've been working towards a big release, which will remove the old `peniko::Color` type and use the new `color` crate instead.
This eventual release will be the start of a path towards wider color gamut support throughout the Linebender ecosystem.
```

File: content/blog/2024-12-03-tmil-11/index.md [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+We have long had a pain point with Peniko where it defines some resource management types, `Blob`, `Font`, and `WeakBlob`.
+These are used within Parley and Vello as well as by anyone using those crates.
+We plan to move these into a separate crate to ease versioning issues and to take advantage of this to improve the functionality.
+This will probably be coming in the next big release following the one adding Color.
+
+## Vello
+
+Vello too will have a big release within the next month, building upon the new releases of Peniko and Color.
+
+Additionally:
```
Comment (xStrom) [inline content/blog/2024-12-03-tmil-11/index.md:62]: More about November, less about the future.

```suggestion
Vello was updated to `wgpu` v23.
```

File: content/blog/2024-12-03-tmil-11/index.md [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+
+***SOMEONE ELSE WRITE THIS***
+
+## Xilem
+
+***SOMEONE ELSE WRITE THIS***
+
+## Piet
+
+While we aren't yet planning on a new release of Piet as there have been no changes since last month, we did begin a discussion about a possible major revision to Piet and what a next-generation Piet might look like.
```
Comment (xStrom) [inline content/blog/2024-12-03-tmil-11/index.md:83]: Talking about the absence of plans is even less useful than talking about plans. Let's cut it.

```suggestion
While there were no code changes, we did begin a discussion about a possible major revision to Piet and what a next-generation Piet might look like.
```

File: content/blog/2024-12-03-tmil-11/index.md [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+We look forward to continuing to discuss this topic over the coming months.
+
+## tiny-skia, resvg, simplecss, svgtypes
+
+In the last month, we have made progress on converting the simplecss and svgtypes crates over to using the Linebender CI infrastructure.
+Work will continue on the tiny-skia and resvg repositories.
+
+The re-licensing of the crates within the resvg repository has been completed and these are now under the "Apache-2.0 OR MIT" license pair.
+
+Additionally, a couple of exciting improvements have landed in the resvg repository:
```
Comment (xStrom) [inline content/blog/2024-12-03-tmil-11/index.md:100]: Capitalization. Also, skip talking about plans of future CI work.

```suggestion
## Tiny Skia, Resvg, SimpleCSS, SVG Types

In the last month, we have made progress on converting the SimpleCSS and SVG Types projects over to using the Linebender CI infrastructure.

The re-licensing of the code within the Resvg repository has been completed and is now under the "Apache-2.0 OR MIT" license pair.

Additionally, a couple of exciting improvements have landed in the Resvg repository:
```

File: content/blog/2024-12-03-tmil-11/index.md [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+
+* [resvg#843][]: Add support for the `!important` attribute.
+* [resvg#850][]: Add support for embedded luma JPEG images.
+
+We haven't yet set a timeline for doing the next release of the resvg family of crates as it will be the first that we have done after assuming stewardship of these crates.
+
+## Velato, Vello SVG
+
+[velato 0.4.0][] was released, updating it to use Vello 0.3. Additionally, 0.3.1 was released which correctly implemented non-linear easing.
```
Comment (xStrom) [inline content/blog/2024-12-03-tmil-11/index.md:109]: One line per sentence.

```suggestion
[Velato 0.4.0][] was released, updating it to use Vello 0.3.
Additionally, 0.3.1 was released which correctly implemented non-linear easing.
```

File: content/blog/2024-12-03-tmil-11/index.md [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+[resvg#843]: https://github.com/linebender/resvg/pull/843
+[resvg#850]: https://github.com/linebender/resvg/pull/850
+
+[vello#722]: https://github.com/linebender/vello/pull/722
+[vello#735]: https://github.com/linebender/vello/pull/735
+[vello#740]: https://github.com/linebender/vello/pull/740
+[vello#743]: https://github.com/linebender/vello/pull/743
+
+[velato 0.4.0]: https://github.com/linebender/velato/releases/tag/v0.4.0
+[vello_svg 0.5.0]: https://github.com/linebender/vello_svg/releases/tag/v0.5.0
```
Comment (xStrom) [inline content/blog/2024-12-03-tmil-11/index.md:158]:
```suggestion
[Velato 0.4.0]: https://github.com/linebender/velato/releases/tag/v0.4.0
[Vello SVG 0.5.0]: https://github.com/linebender/vello_svg/releases/tag/v0.5.0
```

File: content/blog/2024-12-03-tmil-11/index.md [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+While we are focused on shipping code today, we still have an eye on the future and how to be prepared for the new opportunities and technologies that are coming.
+This can be seen in our thinking about the future of Piet as previously discussed as well as the work that we're doing with Color.
+
+We have a couple of active and ongoing research projects:
+
+* Raph Levien on new rendering approaches for Vello.
+* Joe Neeman on boolean path operations.
+* ***IS THERE ANYTHING ELSE TO MENTION HERE?***
+
+This last month also saw the transition of the Slang shader language from Nvidia to Khronos.
```
Comment (xStrom) [inline content/blog/2024-12-03-tmil-11/index.md:125]:
```suggestion
November also saw the transition of the Slang shader language from Nvidia to Khronos.
```

File: content/blog/2024-12-03-tmil-11/index.md (hunk 2 above)
Comment (DJMcNab) [reply content/blog/2024-12-03-tmil-11/index.md:51]: I agree with the first paragraph; I think "eventual" is probably going too far the other way.

Comment (waywardmonkeys) [reply content/blog/2024-12-03-tmil-11/index.md:51]: Edited your suggestion to remove "eventual" ... it isn't eventual. It is happening within the next 2 weeks, probably 1 week.

File: content/blog/2024-12-03-tmil-11/index.md [hunk 9]
```diff
@@ -0,0 +1,158 @@
...
+When we published the update for [October, 2024](/blog/tmix-10/), we had already been discussing starting a new color-focused crate, but hadn't yet created a repository.
+
+We may have a dedicated blog post about the Color crate in the future and laying out some of the motivations, but we'll go over it a bit here.
+
+Linebender, among other things, believes in helping build solid foundations for people building user interfaces and working in graphics, in the Rust programming language (and on GPUs).
+Upon seeing a problem, we prefer to produce a focused crate that solves that problem and that can be used by others within the Rust ecosystem.
+Color is one such area.
+It is much harder than it may appear at first, with many complicating factors like what should be done by a basic color struct, what color spaces to support, and whether or not to support color formats used in print (like CMYK).
+
+We looked around at a wide variety of crates in various stages of maintenance and development, their feature sets, the "weight" of those crates, and how closely they fit in with what we would need in Vello, tiny-skia, resvg, svgtypes, Masonry, and Xilem.
```
[the remark this answers is not in the document]
Comment (tomcur) [reply content/blog/2024-12-03-tmil-11/index.md]: Are we sure about this capitalization (aside from SVG Types)? The original version is as the names were capitalized in [last month's post](https://github.com/linebender/linebender.github.io/blob/f2935b0412098e1ca2075191f46f1f10880a3659/content/blog/2024-11-04-tmix-10/index.md#resvg-tiny-skia-and-related-projects).

[the remark this answers is not in the document]
Comment (xStrom) [reply content/blog/2024-12-03-tmil-11/index.md]: We're still in the process of transitioning the capitalization change for these projects, so yes plenty of old material with old names. I just think we can use the new capitalization here already.

[the remark this answers is not in the document]
Comment (tomcur) [reply content/blog/2024-12-03-tmil-11/index.md]: Agreed, it just confused me for a minute.

File: content/blog/2024-12-03-tmil-11/index.md [hunk 10]
```diff
@@ -0,0 +1,228 @@
...
+
+* [xilem#731][], [xilem#734][]: Added a example using our SVG support for drawing, called `svgdraw`.
+* [xilem#749][]: Experimentally increased support for custom view context types.
+
+## Masonry
+
+Masonry is the widget system used by Xilem.
+It provides a non-opinionated retained widget tree, designed as a base layer for high-level GUI frameworks.
+
+* [xilem#670][]: Added some features needed for custom client side decorations, by  Marco Melorio.
```
Comment (xStrom) [inline content/blog/2024-12-03-tmil-11/index.md:89]:
```suggestion
* [xilem#670][]: Added some features needed for custom client side decorations, by Marco Melorio.
```

File: content/blog/2024-12-03-tmil-11/index.md [hunk 11]
```diff
@@ -0,0 +1,228 @@
...
+* [xilem#735][]: Viktor Strate Kløvedal found and resolved several features which incorrectly private.
```
Comment (xStrom) [inline content/blog/2024-12-03-tmil-11/index.md:90]: This seems confusing. Based on a quick skim of the PR I would write it as:

```suggestion
* [xilem#735][]: Viktor Strate Kløvedal made several widget properties publicly accessible.
```

File: content/blog/2024-12-03-tmil-11/index.md [hunk 12]
```diff
@@ -0,0 +1,228 @@
...
+
+* [kurbo#390][] Reduce number of operations in `Triangle::circumscribed_circle`.
+* [kurbo#399][] Implement `Sum` for `Vec2`.
+
+## Peniko
+
+We've been working towards a big release, which will remove the old `peniko::Color` type and use the new `color` crate instead.
+This release will be the start of a path towards wider color gamut support throughout the Linebender ecosystem.
+
+
```
Comment (xStrom) [inline content/blog/2024-12-03-tmil-11/index.md:134]: Ultimate pedantry. 😂
```suggestion

```

File: content/blog/2024-12-03-tmil-11/index.md [hunk 13]
```diff
@@ -0,0 +1,228 @@
...
+[xilem#731]: https://github.com/linebender/xilem/pull/743
+[xilem#732]: https://github.com/linebender/xilem/pull/743
+[xilem#734]: https://github.com/linebender/xilem/pull/743
+[xilem#735]: https://github.com/linebender/xilem/pull/743
+[xilem#736]: https://github.com/linebender/xilem/pull/743
+[xilem#742]: https://github.com/linebender/xilem/pull/743
+[xilem#749]: https://github.com/linebender/xilem/pull/743
+[xilem#754]: https://github.com/linebender/xilem/pull/743
+[xilem#755]: https://github.com/linebender/xilem/pull/743
+[xilem#762]: https://github.com/linebender/xilem/pull/743
```
Comment (xStrom) [inline content/blog/2024-12-03-tmil-11/index.md:224]: Copy-paste gone wrong here.
```suggestion
[parley#152]: https://github.com/linebender/parley/pull/152
[parley#154]: https://github.com/linebender/parley/pull/154
[parley#156]: https://github.com/linebender/parley/pull/156
[parley#158]: https://github.com/linebender/parley/pull/158
[parley#160]: https://github.com/linebender/parley/pull/160
[parley#166]: https://github.com/linebender/parley/pull/166
[parley#170]: https://github.com/linebender/parley/pull/170
[parley#171]: https://github.com/linebender/parley/pull/171
[parley#173]: https://github.com/linebender/parley/pull/173
[parley#182]: https://github.com/linebender/parley/pull/182
[parley#184]: https://github.com/linebender/parley/pull/184
[resvg#843]: https://github.com/linebender/resvg/pull/843
[resvg#850]: https://github.com/linebender/resvg/pull/850
[vello#722]: https://github.com/linebender/vello/pull/722
[vello#735]: https://github.com/linebender/vello/pull/735
[vello#740]: https://github.com/linebender/vello/pull/740
[vello#743]: https://github.com/linebender/vello/pull/743
[xilem#670]: https://github.com/linebender/xilem/pull/670
[xilem#731]: https://github.com/linebender/xilem/pull/731
[xilem#732]: https://github.com/linebender/xilem/pull/732
[xilem#734]: https://github.com/linebender/xilem/pull/734
[xilem#735]: https://github.com/linebender/xilem/pull/735
[xilem#736]: https://github.com/linebender/xilem/pull/736
[xilem#742]: https://github.com/linebender/xilem/pull/742
[xilem#749]: https://github.com/linebender/xilem/pull/749
[xilem#754]: https://github.com/linebender/xilem/pull/754
[xilem#755]: https://github.com/linebender/xilem/pull/755
[xilem#762]: https://github.com/linebender/xilem/pull/762
```

File: content/blog/2024-12-03-tmil-11/index.md [hunk 14]
```diff
@@ -0,0 +1,228 @@
...
+Vello is our GPU vector renderer.
+It can draw large 2D scenes with high performance, using GPU compute shaders for most of the work.
+
+* [vello#735][], [vello#743][]: Update to wgpu 23.0.1, matching what is used by Bevy 0.15.
+* [vello#722][]: Improving image sampling.
+* [vello#740][]: Correcting support for PNG glyphs in fonts, important for emoji on macOS.
+
+## Parley
+
+Parley is a text layout library. It handles text layout, mostly at the level of line breaking and resolving glyph positions.
```
Comment (xStrom) [inline content/blog/2024-12-03-tmil-11/index.md:109]: One line per sentence.
```suggestion
Parley is a text layout library.
It handles text layout, mostly at the level of line breaking and resolving glyph positions.
```

**Review by xStrom — COMMENTED** [review]
Looks good overall, but found some room for improvement.

File: content/blog/2024-12-03-tmil-11/index.md (hunk 13 above)
Comment (DJMcNab) [reply content/blog/2024-12-03-tmil-11/index.md:224]: Wow... That's an impressive error! Thanks

File: content/blog/2024-12-03-tmil-11/index.md [hunk 15]
```diff
@@ -0,0 +1,227 @@
...
+Kurbo has seen some minor updates within the last month and we're planning to publish them in a minor patch release within the next month.
+
+* [kurbo#390][] Reduce number of operations in `Triangle::circumscribed_circle`.
+* [kurbo#399][] Implement `Sum` for `Vec2`.
+
+## Peniko
+
+We've been working towards a big release, which will remove the old `peniko::Color` type and use the new `color` crate instead.
+This release will be the start of a path towards wider color gamut support throughout the Linebender ecosystem.
+## Piet
```
Comment (xStrom) [inline content/blog/2024-12-03-tmil-11/index.md:134]: Haha, my goal was to get a single empty line here to match the other sections. 😅

```suggestion
This release will be the start of a path towards wider color gamut support throughout the Linebender ecosystem.

## Piet
```

File: content/blog/2024-12-03-tmil-11/index.md [hunk 16]
```diff
@@ -0,0 +1,227 @@
...
+This is the reference for the above gradient, and should appear identical.
+
+</figcaption>
+</figure>
+
+Since then, we have begun the work of making Peniko and Vello use the Color crate and improving it to support that usage.
+Early next year, we hope to use it within SVG Types, Resvg, and Tiny Skia.
+
+We will be publishing a 0.2.0 release within the next month and encourage people to give it a try.
+The 0.2.0 release will include some additional (non-CSS 4) color spaces like ACEScg, OkHsl, and OkHsv as well as things that we found useful when using it in Peniko and Vello.
```
Comment (tomcur) [inline content/blog/2024-12-03-tmil-11/index.md:61]: OkHsl and OkHsv didn't make the release.

```suggestion
The 0.2.0 release will include the (non-CSS 4) color space ACEScg, as well as things that we found useful when using it in Peniko and Vello.
```

Comment (DJMcNab) [reply content/blog/2024-12-03-tmil-11/index.md]: This is part of why I didn't like us being specific about "what we'll do in December", when it turns out not to be what we'll do.

**Review by xStrom — APPROVED** [review]
Looks good!

File: content/blog/2024-12-03-tmil-11/index.md [hunk 17]
```diff
@@ -0,0 +1,228 @@
...
+
+The majority of changes in November were made in the [widget layer](#masonry), which has its own section.
+This month only had small updates otherwise:
+
+* [xilem#742][]: Made all built-in `View` types `#[must_use]`, improving error reporting in cases where views could be silently ignored.
+* [xilem#732][]: Implemented `View` for `Rc<impl View>`.
+
+Work on Xilem web has been continuing well:
+
+* [xilem#731][], [xilem#734][]: Added a example using our SVG support for drawing, called `svgdraw`.
```
Comment (tomcur) [inline content/blog/2024-12-03-tmil-11/index.md:81]:
```suggestion
* [xilem#731][], [xilem#734][]: Added an example using our SVG support for drawing, called `svgdraw`.
```

**Review by tomcur — APPROVED** [review]

**Pull request merged**